storage_v2/v1adapter/writer.go

```diff
@@ -13,10 +13,19 @@ import (
 	"github.com/jaegertracing/jaeger/storage_v2/tracestore"
 )
 
+var errV1WriterNotAvailable = errors.New("spanstore.Writer is not a wrapper around v1 writer")
+
 type TraceWriter struct {
 	spanWriter spanstore.Writer
 }
 
+func GetV1Writer(writer tracestore.Writer) (spanstore.Writer, error) {
```

Comment:
if the only reason you need this is to access the Close method of v1 writer then just add Close to TraceWriter instead